### PR TITLE
Add build script

### LINKED ISSUE #404: Use ES2015 modules when available
https://twitter.com/addyosmani/status/859296190323597313

### PR DESCRIPTION
**Pull request by bfred-it**
Untested. Pushing it here so you can take a look.

Closes #404 not found.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,11 +1,18 @@
 {
   "scripts": {
     "test": "xo && ava",
+    "build": "cd src && each '{*.js,!(libs)/**/*.js}' --parallel 10 -- rollup --config node:webext $PWD/%p --output $PWD/../extension/%p",
```
Comment (sindresorhus) [inline package.json]: `--parallel 10` => `--parallel=4` I'm not seeing any better perf after 4. Actually slightly less.

`!(libs)/**/*.js` => `!libs/vendor/*.js` to be explicit

File: package.json [hunk 2]
```diff
@@ -25,6 +35,22 @@
...
+    "require": [
+      "babel-register"
+    ]
+  },
+  "babel": {
+    "plugins": [
+      "transform-es2015-modules-commonjs"
     ]
+  },
+  "dependencies": {
```
Comment (sindresorhus) [inline package.json]: Put this above `devDependencies`

File: package.json (hunk 1 above)
Comment (bfred-it) [inline package.json]: It's actually more about running it with `watch` than concurrency. If `--parallel=4` and the files are 5, the 5th one won't be compiled.

Comment (bfred-it) [inline package.json]: I'm using `each` to create a copy of all the files in `src` to `extension` without having to manually configure each new file. The `!(libs)` is needed to skip the submodules we have, not just vendor files. 

Improvements to the glob are welcome though.

Comment (sindresorhus) [inline package.json]: But there are more than 10 files. This seems like weird constraint. Any better way we could solve this?

**Comment by sindresorhus** [comment]
I'm getting:

screen shot 2017-06-02 at 12 55 30

File: package.json (hunk 1 above)
Comment (bfred-it) [inline package.json]: No no the files that `each` is looking for is are currently just three. Content, background and options. The "clean" alternative 3 separate build steps or an alternative looping tool that just processes as many files as it finds. Unfortunately rollup does not support multiple builds in one call

Comment (bfred-it) [inline package.json]: Have a look at 559b71fb78c223dd9a89d0fc26f4f521af252afd for the "clean" alternative

**Comment by hkdobrev** [comment]
image

image

🤣

**Comment by bfred-it** [comment]
[Never](https://www.youtube.com/watch?v=dQw4w9WgXcQ)

**Comment by bfred-it** [comment]
@sindresorhus The jQuery issue was fixed.

I hadn't imported jQuery because I thought it was used by multiple files and didn't want it to be bundled multiple times — but it's only used by `content.js` so we're good. Leaving it as a global script didn't feel right at this point because it involved de-gitignoring it.

The extension now loads correctly and it seems to be working. I haven't tested every single function but it seems to be alright everywhere I looked.

**Comment by sindresorhus** [comment]
I've been testing it for a bit and couldn't find any regressions. @hkdobrev @ Can you also test?

**Comment by sindresorhus** [comment]
This is really awesome @bfred-it! ✨🦄

**Comment by sindresorhus** [comment]
happy

**Comment by bfred-it** [comment]
Excellent! I'll definitely be using it as a base now

**Pull request merged**